lms/djangoapps/course_home_api/progress/v1/views.py

```diff
@@ -121,17 +130,34 @@ def get(self, request, *args, **kwargs):
         monitoring_utils.set_custom_attribute('is_staff', request.user.is_staff)
         is_staff = bool(has_access(request.user, 'staff', course_key))
 
-        _, request.user = setup_masquerade(
-            request,
-            course_key,
-            staff_access=is_staff,
-            reset_masquerade_data=True
-        )
-
-        course = get_course_with_access(request.user, 'load', course_key, check_if_enrolled=False)
+        if student_id is None or student_id == request.user.id:
```

Comment:
this PR is mostly copying over the logic from the old progress page which handles this case that was not ported over

Reply:
Feels weird to me that we do masquerading if you provide your own staff student id. But I do agree that's how the old code worked.